erpnext/stock/doctype/item/item.py

```diff
@@ -898,6 +905,77 @@ def validate_auto_reorder_enabled_in_stock_settings(self):
 	def validate_valuation_method(self):
 		if self.is_new() and not self.valuation_method:
 			self.valuation_method = frappe.db.get_single_value("Stock Settings", "valuation_method")
+	
+	def update_to_bloomtrace(self):
+		"""
+		Create a new item on Bloomtrace via Bloomstack.
+		Args: 
+		
+		Returns: 
+			Successful Message to the User
+		"""
+		try:
+			now = datetime.now()
+			request_data = {
+				"site_url": "manufacturing.bloomstack.io",
+				"customer_name": "Bloomstack",
+				"company_name": "Bloomstack India",
+				"license_number": "A12-0000015-LIC",
+				"rest_method": "POST",
+				"environment": "sandbox",
+				"doctype": "Item",
+				"doctype_data": {
+					"name": self.metrc_item_name,
+					"item_category": self.metrc_item_category,    
+					"uom": self.metrc_uom,
+					"creation": now.strftime("%Y-%m-%dT%H:%M%z"),    
+					"modified": now.strftime("%Y-%m-%dT%H:%M%z"),
+					"modified_by": "neil@bloomstack.com",
+					"parent": None,
+					"parentfield": None,
+					"parenttype": None,
+					"idx": 0,
+					"docstatus": 0,
+					"company": "Bloom91"
+				}
+			}
+
+			# for item_category which requires strain
+			if self.strain_name:
+				request_data['doctype_data']['strain'] = self.strain_name
+				
+			# getting the item category from the Compliance Item Category for quantity type and mandatory unit
+			item_category = frappe.db.get("Compliance Item Category", self.metrc_item_category, ['quantity_type', 'mandatory_unit'])
+
+			# setting the unit uom and unit value as per the countbased quantity type
+			if item_category.get('quantity_type') == "CountBased":
```

Comment:
any other quantity types that need to be handled?

Reply:
No, there are 3 quatity types but unit value and unit uom the countbased quantity is the required quantity, and there is no other senarios for any other category.